federation/executor_test.go

```diff
@@ -453,6 +503,111 @@ func TestExecutorQueriesBasic(t *testing.T) {
 	}
 }
 
+// This function checks that we are able to query for and receive an object that was passed between gql servers
+// Specifically, it tests the functions in planner_helpers.go
+func TestExecutorQueriesWithObjectKey(t *testing.T) {
```

Comment:
I think the test does not verify that we are passing object as keys since `s1` is able to resolve all the fields. 
I think we want to remove the `UserSkills` from `s1` schema. 
Also, I would suggest to add a test in planner_test.go to directly verify the planner output.